### PR TITLE
Don't overwrite directory permissions on --chown

### PR DESCRIPTION
**Pull request by bertinatto**
In a previous PR ( #336) I failed to check if my changes were changing the ownership of:

1) existent target directory
2) existing files in the target directory

As a result, I'm proposing these changes to fix that. 

I'm happy to re-work my patch if you think the approach isn't appropriate.

Thanks.

### CODE REVIEW

**Comment by nalind** [comment]
bot, add author to whitelist

File: add.go [hunk 1]
```diff
@@ -210,26 +210,45 @@ func (b *Builder) user(mountPoint string, userspec string) (specs.User, error) {
...
+}
+
+// ensureDir creates a directory if it doesn't exist, setting ownership and permissions as passed by user and perm.
+func ensureDir(path string, user specs.User, perm os.FileMode) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.MkdirAll(path, perm); err != nil {
+			return errors.Wrapf(err, "error ensuring directory %q exists", path)
+		}
+		if err := os.Chown(path, int(user.UID), int(user.GID)); err != nil {
+			return errors.Wrapf(err, "error ensuring directory %q exists", path)
```
Comment (rhatdan) [inline add.go]: SHouldn't this be
"error setting ownership of %q"

File: add.go [hunk 2]
```diff
@@ -154,14 +154,14 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 				// to create it first, so that if there's a problem,
 				// we'll discover why that won't work.
 				d := dest
-				if err := os.MkdirAll(d, 0755); err != nil {
```
Comment (rhatdan) [inline add.go]: Nit, should just remove `d` and use `dest` everywhere.

File: add.go (hunk 1 above)
Comment (bertinatto) [reply add.go]: Thanks for reviewing!

**Comment by rhatdan** [comment]
LGTM
@nalind PTAL

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**